### PR TITLE
Allow enabling low-level debugging of fwupdx64.efi from uefi_capsule.…

### PR DESCRIPTION
**Pull request by hughsie**
…conf

Short-to-medium-term we want to stop installing fwupdate and this was the last useful part we never migrated.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/uefi-capsule/uefi_capsule.conf [hunk 1]
```diff
@@ -1,5 +1,8 @@
 [uefi_capsule]
 
+# enable the low-level debugging of fwupdx64.efi to the FWUPDATE_DEBUG_LOG EFI variable
```
Comment (superm1) [inline plugins/uefi-capsule/uefi_capsule.conf:3]: Perhaps we want a warning here that this is going to fill up the NVRAM store more quickly and should only be enabled for working on an issue?

File: plugins/uefi-capsule/fu-uefi-nvram-device.c [hunk 2]
```diff
@@ -40,6 +73,10 @@ fu_uefi_nvram_device_get_results(FuDevice *device, GError **error)
 		return TRUE;
 	}
 
+	/* capture EFI binary debug output */
+	if (fu_device_has_private_flag(device, FU_UEFI_DEVICE_FLAG_ENABLE_DEBUGGING))
+		fu_uefi_nvram_device_capture_debugging(device);
```
Comment (superm1) [inline plugins/uefi-capsule/fu-uefi-nvram-device.c:78]: I would think we want this even when GRUB is used too not just NVRAM.

File: plugins/uefi-capsule/uefi_capsule.conf (hunk 1 above)
Comment (superm1) [reply plugins/uefi-capsule/uefi_capsule.conf:3]: Also we need to mention it's not used for CoD

File: plugins/uefi-capsule/fu-uefi-device.h [hunk 3]
```diff
@@ -84,6 +84,12 @@ typedef enum {
  * Do not prepend a plausible missing capsule header.
  */
 #define FU_UEFI_DEVICE_FLAG_NO_CAPSULE_HEADER_FIXUP (1 << 7)
+/**
+ * FU_UEFI_DEVICE_FLAG_ENABLE_DEBUGGING:
+ *
+ * Increase debugging level for the device.
+ */
+#define FU_UEFI_DEVICE_FLAG_ENABLE_DEBUGGING (1 << 8)
```
Comment (superm1) [inline plugins/uefi-capsule/fu-uefi-device.h:92]: `FU_UEFI_DEVICE_FLAG_ENABLE_EFI_DEBUGGING` perhaps

File: plugins/uefi-capsule/fu-uefi-device.h [hunk 4]
```diff
@@ -84,6 +84,12 @@ typedef enum {
  * Do not prepend a plausible missing capsule header.
  */
 #define FU_UEFI_DEVICE_FLAG_NO_CAPSULE_HEADER_FIXUP (1 << 7)
+/**
+ * FU_UEFI_DEVICE_FLAG_ENABLE_DEBUGGING:
+ *
+ * Increase debugging level for the device.
```
Comment (superm1) [inline plugins/uefi-capsule/fu-uefi-device.h:90]: `Enable debugging the EFI binary` perhaps

Comment (hughsie) [reply plugins/uefi-capsule/fu-uefi-device.h:90]: WFM, thanks.

**Comment by superm1** [comment]
> Short-to-medium-term we want to stop installing fwupdate and this was the last useful part we never migrated.

I think as part of this PR you should tear out the logging functionality from `fwupdate` so that it can't conflict with doing it here.

Either this PR or another mark `fwupdate` as noinst.

File: plugins/uefi-capsule/fu-uefi-device.h (hunk 3 above)
Comment (hughsie) [reply plugins/uefi-capsule/fu-uefi-device.h:92]: Yup, that's better.

File: plugins/uefi-capsule/fu-uefi-nvram-device.c (hunk 2 above)
Comment (hughsie) [reply plugins/uefi-capsule/fu-uefi-nvram-device.c:78]: Ack, good catch. Fixed.

File: plugins/uefi-capsule/uefi_capsule.conf (hunk 1 above)
Comment (hughsie) [reply plugins/uefi-capsule/uefi_capsule.conf]: Something like this?

**Review by superm1 — APPROVED** [review]

**Pull request merged**